src/fs.rs

```diff
@@ -1,13 +1,42 @@
+extern crate rand;
+
 use std;
-use std::fs::File;
+use std::fs::{create_dir, File};
+use std::io::ErrorKind;
 use std::io::Write;
-use std::path::Path;
+use std::path::{Path, PathBuf};
+
+use self::rand::{thread_rng, RngCore};
 
 pub fn write_file_sync(path: &Path, content: &[u8]) -> std::io::Result<()> {
   let mut f = File::create(path)?;
   f.write_all(content)
 }
 
+pub fn make_temp_dir(
+  dir: Option<&Path>,
+  prefix: Option<&str>,
+  suffix: Option<&str>,
+) -> std::io::Result<PathBuf> {
+  let prefix_ = prefix.unwrap_or("");
+  let suffix_ = suffix.unwrap_or("");
+  let mut buf: PathBuf = match dir {
+    Some(ref p) => p.to_path_buf(),
+    None => std::env::temp_dir(),
+  }.join("_");
+  loop {
+    let unique = thread_rng().next_u32();
+    buf.set_file_name(format!("{}{:08x}{}", prefix_, unique, suffix_));
```

Comment:
We have this `tempfile` crate already built into Deno... why not use that? https://docs.rs/tempfile/3.0.3/tempfile/struct.TempDir.html

If we don't want to use that, can we eliminate it and only use this function?

Reply:
Tempfile deletes the directory when it goes out of scope.
I looked into it but it's of no use.